it/test_utils/ResponseHelpers.scala

```diff
@@ -0,0 +1,22 @@
+package test_utils
+
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder
+import com.github.tomakehurst.wiremock.client.WireMock.aResponse
+
+trait ResponseHelpers {
+  def badGateway(): ResponseDefinitionBuilder = {
+    aResponse().withStatus(502)
+  }
+
+  def closedCopeWorkItem(): ResponseDefinitionBuilder = {
+    aResponse().withStatus(422).withBody("CLOSED_COPE_WORK_ITEM")
+  }
+
+  def gatewayTimeout(): ResponseDefinitionBuilder = {
+    aResponse().withStatus(504)
+  }
```

Comment:
In a case of a timeout the http client never received any answer from the server.
Here you simply returns a status 504 to the http client which is different.

To test for a timeout you need something like this for your stub:
`        server.stubFor(get(urlPathEqualTo("/rest/auth/1/session")).willReturn(
          aResponse()
            .withStatus(200)
            .withBody("")
            .withFixedDelay(25000)
        )
        )`
        
        The client timeout at 20s by default so it significantly slow down your tests. You can change that by changing the timeout setting in application.conf for your tests. Create the file tests/resources/application.conf with the content:
`include "/conf/application.conf"

play.ws.timeout.request = "1000ms"
play.ws.timeout.connection = "500ms"`

Look in the live-incident-support-tooll for some context

Reply:
Thanks for the thorough explanation Pascal. I see what you mean. I'll work on those changes.